pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +333,86 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{})
+	registerHandlers(c, n.gatewayResourceClient, "Gateways", n.handleGatewayResource, nil)
+	n.gatewayResourceClient.Start(stop)
+}
+
+// handleGateway resource adds a NetworkGateway for each combination of address and auto-passthrough listener
+// discovering duplicates from the generated Service is not a huge concern as we de-duplicate in NetworkGateways
+// which returns a set, although it's not totally efficient.
+func (n *networkManager) handleGatewayResource(_ controllers.Object, obj controllers.Object, event model.Event) error {
+	gw, ok := obj.(*v1beta1.Gateway)
+	if !ok {
+		return nil
+	}
+
+	if nw := gw.GetLabels()[label.TopologyNetwork.Name]; nw == "" {
+		return nil
+	}
+
+	// we don't want to discover gateways with class "istio-remote" from outside cluster's API servers.
+	// this will only pass for the "config cluster" or "local cluster"
+	if !n.discoverRemoteGatewayResources && gw.Spec.GatewayClassName == "istio-remote" {
```

Comment:
nit: use constants.RemoteGatewayClassName

Actual useful comment: I am a bit worried about tying this to the class.

We have a few use cases I think:
* I want to represent or expose a EW-GW as a service (today)
* I want to _represent_ a EW-GW as a Gateway (this PR)

But is there also a case where we want to actually _deploy_ a EW GW and use it as the cluster gateway? For example, make `class: aws-l4-external` or something? Would a user then have to make another `istio-remote` Gateway to access that? it feels a bit off

Reply:
Actually I see this is only for config cluster

Reply:
This is when we specifically _don't_ want to deploy and if we don't want to deploy, we should only discover locally.  hope that makes sense.